doc/runtime-information.rst

```diff
@@ -4,46 +4,60 @@ Run-time Information
 =====================
 
 Your app should run in a bundle exactly as it does when run from source.
-However, you may need to learn at run-time
-whether the app is running from source, or is "frozen" (bundled).
-For example, you might have
-data files that are normally found based on a module's ``__file__`` attribute.
-That will not work when the code is bundled.
+However, you may want to learn at run-time whether the app is running from
+source or whether it is bundled. You can use the following code to check "are
+we bundled?"::
 
-The |PyInstaller| |bootloader| adds the name ``frozen`` to the ``sys`` module.
-So the test for "are we bundled?" is::
+    import sys
+    if getattr(sys, 'frozen') and hasattr(sys, '_MEIPASS'):
+        print('running in a PyInstaller bundle')
+    else:
+        print('running in a normal Python process')
 
-	import sys
-	if getattr( sys, 'frozen', False ) :
-		# running in a bundle
-	else :
-		# running live
+When a bundled app starts up, the |bootloader| sets the ``sys.frozen``
+attribute and stores the absolute path to the bundle folder in
+``sys._MEIPASS``. For a one-folder bundle, this is the path to that folder. For
+a one-file bundle, this is the path to the temporary folder created by the
+|bootloader| (see :ref:`How the One-File Program Works`).
 
-When your app is running, it may need to access data files in any of
-three general locations:
+When your app is running, it may need to access data files in one of the
+following locations:
 
 * Files that were bundled with it (see :ref:`Adding Data Files`).
-
 * Files the user has placed with the app bundle, say in the same folder.
-
 * Files in the user's current working directory.
 
-The program has access to several path variables for these uses.
+The program has access to several variables for these uses.
+
+
+Using ``__file__``
+~~~~~~~~~~~~~~~~~~
+
+When your program is not bundled, the Python variable ``__file__`` refers to
+the current path of the module it is contained in. When importing a module
+from a bundled script, the |PyInstaller| |bootloader| will set the module's
+``__file__`` attribute to the correct path relative to the bundle folder.
+
+For example, if you import ``mypackage.mymodule`` from a bundled script, then
+the ``__file__`` attribute of that module will be ``sys._MEIPASS +
+'mypackage/mymodule.pyc'``.  So if you have a data file at
+``mypackage/file.dat`` that you added to the bundle at ``mypackage/file.dat``,
+the following code will get its path (in both the non-bundled and the bundled
+case)::
 
+    from os import path
+    path_to_dat = path.join(path.dirname(__file__), 'file.dat')
 
-Using ``__file__`` and ``sys._MEIPASS``
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Why change the headline?

Reply:
One "wants" to use `__file__` and not `sys._MEIPASS` (although one might have to in some edge cases). I kind of wanted to remove the focus from `sys._MEIPASS`